docs/src/main/sphinx/routines/function.md

```diff
@@ -0,0 +1,95 @@
+# FUNCTION
+
+## Synopsis
+
+```text
+FUNCTION name ( [ parameter_name data_type [, ...] ] )
+  RETURNS type
+  [ LANGUAGE langauge]
+  [ NOT? DETERMINISTIC ]
+  [ RETURNS NULL ON NULL INPUT ]
+  [ CALLED ON NULL INPUT ]
+  [ SECURITY { DEFINER | INVOKER } ]
+  [ COMMENT description]
+  statements
+```
+
+## Description
+
+Declare a SQL routine.
+
+The `name` must qualify the name of the catalog to store the routine, unless the
+routine is only declared in the scope of a query.
+
+The list of parameters is a comma-separated list of names `parameter_name` and
+data types `data_type`, see [data type](/language/types). An empty list, specified as
+`()` is also valid.
+
+The `type` value after the `RETURNS` keyword identifies the [data
+type](/language/types) of the routine output.
+
+The optional `LANGUAGE` characteristic identifies the language used for the
+routine definition with `language`. Only `SQL` is supported.
+
+The optional `DETERMINISTIC` or `NOT DETERMINISTIC` characteristic declares that
+the routine is deterministic. This means that repeated routine calls with
+identical input parameters yield the same result. For SQL language routines, a
+routine is non-deterministic if it calls any non-deterministic routines and
+[functions](/functions).
+
+The optional `RETURNS NULL ON NULL INPUT` characteristic declares that the
+routine returns a `NULL` value when any of the input parameters are `NULL`.
+The routine is not invoked with a `NULL` input value.
+
+The `CALLED ON NULL INPUT` characteristic declares that the routine is invoked
+with `NULL` input parameter values.
```

Comment:
These two are mutually exclusive.  The default is `CALLED ON NULL INPUT`, but most functions should use `RETURNS NULL ON NULL INPUT` unless the code has some special handling for nulls.

Reply:
Added here and on recommendations.